lib/http-bridge/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -439,6 +441,12 @@ fixtureWallet ctx@(Context _ _ _ faucet) = do
             then return (getFromResponse id r)
             else threadDelay oneSecond *> checkBalance wid
 
+-- | One million ADA, in Lovelace, just like this.
+oneMillionAda :: Natural
+oneMillionAda = ada (1_000_000)
+  where
+    ada = (*) (1_000_000)
```

Comment:
I wonder if it would be possible to use the type system here, to distinguish amounts (in Ada or Lovelace) from other values that happen to be `Natural`?

In particular, we're trying to encode something that is a number of `Lovelace`: the number of lovelace in one Ada.

Reply:
We could use `Quantity` etc ... But we purposely use _raw_ types in the integration DSL to keep the test scenarios readable and simple to write.

Reply:
that's was my question : why not `Quantity` here?

Reply:
As just explained, to avoid having to perform to many transformations when writing the integration test scenarios. Scenarios are mostly just about submitting things to the API and asserting results; Manipulating raw types directly there makes it very easy to write these scenarios and piggy-back on the API to do the type-checking etc ... 

More importantly, it also allows us to provide invalid and out-of-range values and still submit them to test that API fulfill its role! This is also why we don't use a typed client in the integration tests. We do want to be able to submit invalid values :)

Reply:
Besides, `Quantity` might not always help. Consider

```haskell
pay :: Quantity "lovelace" Natural -> result

-- later

do
pay (Quantity 100)
...
```

The type is inferred at the call-site and can easily be confused. Would be better to enforce it to be explicit.

Like
```haskell
do
pay (Lovelace 100)
...
```
(I'm sure there are fancier ways too)